libraries/AP_HAL_Linux/Util.cpp

```diff
@@ -62,12 +62,23 @@ void Util::commandline_arguments(uint8_t &argc, char * const *&argv)
     argv = saved_argv;
 }
 
+uint64_t Util::get_hw_rtc() const
+{
+    struct timespec ts;
+    clock_gettime(CLOCK_REALTIME, &ts);
+    const uint64_t seconds = ts.tv_sec;
+    const uint64_t nanoseconds = ts.tv_nsec;
+    return (seconds * 1000000ULL + nanoseconds/1000ULL);
+}
+
 void Util::set_hw_rtc(uint64_t time_utc_usec)
 {
+// don't reset the HW clock time on people's laptops.
```

Comment:
That is interesting!
I wonder if it could have some side effects on our testing using the linux hal ? That isn't related to this PR but nice to notice it